src/tests/test_meta_blk_mgr.cpp

```diff
@@ -614,6 +726,45 @@ TEST_F(VMetaBlkMgrTest, single_read_test) {
     this->shutdown();
 }
 
+TEST_F(VMetaBlkMgrTest, register_dependency_test) {
+    reset_counters();
+    m_start_time = Clock::now();
+    this->register_client_inlcuding_dependencies();
+
+    // simulate reboot case that MetaBlkMgr will scan the disk for all the metablks that were written;
+    this->scan_blks();
+
+    this->recover_with_on_complete();
+
+    std::unordered_map< meta_sub_type, int > subtypes_order_after_sort_for_cb_map;
+
+    // verify the order of callback
```

Comment:
Suggesting moving https://github.com/eBay/HomeStore/pull/200/files#diff-820453c6d78ac456e2f86d1c2960472e447e98c0311c9392253b663e68f43989R633-R647 to test cases.

Better to enhance the checks as we want to ensure all A are before all B/C, i.e last A < first B. The implementation right now doesnt works if we add UT to have more than one records per subsys.

Reply:
good suggestion, will enhance the checks